src/pyglottolog/api.py

```diff
@@ -93,46 +124,65 @@ def glottocodes(self):
     def ftsindex(self):
         return self.build_path('whoosh')
 
+    @lazyproperty
+    def _tree_dirs(self):
+        return list(walk(self.tree, mode='dirs'))
+
     def languoid(self, id_):
+        """
+        Retrieve a languoid.
+
+        :param id_: Glottocode or ISO code.
+        :return: `languoids.Languoid` instance.
+        """
```

Comment:
WDYT about using type annotations instead` (see also 'set of...' below)?

Reply:
Hm, since this is >=py3.5 we might as well. And you are right, for containers that seems even more useful.

Reply:
Reading up and playing around with type hints, I think I like it, in particular in cases as above, where it can fully replace the somewhat generic documentation.

It leaves me thinking, though, that stuff like duck typing might be a code smell - because you can only annotate this as `typing.Any` - right? And is type hinting also useful if it can only be done for selected callables?

Reply:
Yes, I think type hinting is useful even if it is only used selectively. I think duck typing is still fine, you can also use `typing.TypeVar` for things like `takes an iterable and yields pairs of items of that type`. [pytype](https://google.github.io/pytype/user_guide.html)  might be interesting.